### ISSUE TITLE
Easy-to-manage 360-degree rotation

### ISSUE THREAD

**Issue by misaki-eymard** [EsotericSoftware/spine-editor#811]
Currently, if you want to offset the looping 360-degree rotation animation, you should reset the rotation value in a short time so that the first and last key has the same key value, as shown in the following animation:
![1719904600-451130-screen-recording-2024-07-02-at-161508](https://github.com/EsotericSoftware/spine-editor/assets/85478846/090f0c33-5485-4196-9127-f81c13fc7d39)

It would be easier to have an option for forced rotations (e.g. "1 x clockwise", "2 x counterclockwise") to create perfect cycles without having to add subframe keyframes. Perhaps properties similar to sequence attachments would be ideal if this option were added:
![rotation-ui](https://github.com/EsotericSoftware/spine-editor/assets/85478846/ee32c543-38cc-426f-8343-8ff83ba65d3f)

Related forum thread:
https://esotericsoftware.com/forum/d/26453-360-rotation-cycle-not-able-to-use-offset-

**Comment by NathanSweet** [comment]
The `-0.1` stepped key is logical and not so bad, except that it can cause problems when scaling the whole animation. If not holding `shift`, the key will be lost. If holding `shift` it will be OK, but then other keys may land on fractional frames.

This is a general problem with scaling animations where you want some keys on fractional frames and some not. It could make sense to improve scaling, for example to preserve keys off by `0.1` when shift is not hold and they would overwrite a key with different values.

The proposed keyed rotation property is interesting, but makes for a second way to key rotation. We would need to resolve that somehow.

**Comment by FabianoIlCapo** [comment]
True, improving the scaling sounds useful to not bump in the auto snapping-while-scaling problem. It would make sense that fractional keyframes would keep their fractionality while the animation being scaled. They could keep the previous and next snapping keyframes as reference to know where to place themselves.

**Comment by NathanSweet** [comment]
To improve scaling, we end up with 3 ways to perform rounding:

1. Round frames to whole numbers, but only if they were already on whole numbers. If not, no rounding. However, if the key was within 1 frame of the next or previous frame, preserve that distance. This last part keeps eg a key that is 0.1 frame from the next/previous key.
2. No rounding for any keys.
3. Rounding all keys.

Currently we don't have 1 and `shift` selects 2 or 3.

I propose: 1 is the default. Hold `shift` to get 2. Then if `shift` was ever pressed during the drag you get 3.

It's a little bit hidden, but most drags won't use `shift` and get 1, drags that do use `shift` probably want 2, and on the rare occasion you want 3 you probably press `shift` trying to get it. Confusion could result if you want 1 but press `shift` (why would you do that) and then only get 3 OR if you want 3 (rare) and don't know how to get it.

4.3.21-beta
